### DISCUSSION TITLE
Detected blocking call to import_module inside the event loop.

### DISCUSSION THREAD

**Discussion by kozobera** [nathanmarlor/foxess_modbus#617] [General]
Hi Nathan,

Thank you for your excellent work! I just updated HA to version Core 2024.5.4, Frontend 20240501.1 running in Docker. Unfortunately, I found in the log this warning - see below. I don't know enough to identify if this warning applies to my installation only (probably not). Can you please check and advise how to solve it?

Thank you.

Copy/Paste from log:

Logger: homeassistant.util.loop
Source: util/loop.py:99
First occurred: 11:52:49 (3 occurrences)
Last logged: 11:52:49

Detected blocking call to import_module inside the event loop by custom integration 'foxess_modbus' at custom_components/foxess_modbus/client/modbus_client.py, line 203: if auto_connect and hasattr(self._client, "connected") and not self._client.connected: (offender: /usr/local/lib/python3.12/site-packages/serial/__init__.py, line 74: importlib.import_module(package_name)), please create a bug report at https://github.com/nathanmarlor/foxess_modbus/issues
Detected blocking call to import_module inside the event loop by custom integration 'foxess_modbus' at custom_components/foxess_modbus/client/modbus_client.py, line 203: if auto_connect and hasattr(self._client, "connected") and not self._client.connected: (offender: /usr/local/lib/python3.12/site-packages/serial/__init__.py, line 75: handler_module = importlib.import_module(module_name, package_name)), please create a bug report at https://github.com/nathanmarlor/foxess_modbus/issues

**Comment by canton7** [comment]
Thanks. Opened #618

**Reply by canton7** [reply]
Fixed
